qiskit/circuit/bit.py

```diff
@@ -41,13 +41,14 @@ def __init__(self, register, index):
 
         self.register = register
         self.index = index
+        self._hash = hash((self.register, self.index))
```

Comment:
Here and for Register, can we update the hash on attribute changes?

Reply:
Done for bits in: https://github.com/Qiskit/qiskit-terra/pull/3940/commits/59352653ee77f9452ba1c6a8a0d82aee28b6893f

Reply:
Registers too in: https://github.com/Qiskit/qiskit-terra/pull/3940/commits/69da7734c2020c1dab4e75208be291ed13ce53ff